### ISSUE TITLE
How to implement custom multiclass objective function?

### ISSUE THREAD

**Issue by bzografov** [catboost/catboost#856]
Problem: How to implement custom multiclass objective function?
catboost version: 0.12.2
Operating System: Windows 10
CPU

Hi, first of all congratulations for the great library!

I am trying to utilize custom objective function for multiclassification following your guides in:
https://catboost.ai/docs/concepts/python-usages-examples.html#custom-objective-function 

To see how it works, I tried to reproduce the MultiClass loss function, but with the defined gradient and Hessian matrix the program starts and gets stuck without throwing any errors. If I don't define a Hessian (return []), the program runs without problems. Could you see if I am doing anything wrong and let me know what is the expected format of the Hessian matrix.

```python
import numpy as np
from catboost import Pool, CatBoostClassifier


class MultiClassObjective(object):
    def calc_ders_multi(self, approxes, targets, weights):
        # approxes, targets, weights are indexed containers of floats
        # (containers with only __len__ and __getitem__ defined).
        # weights parameter can be None.
        n_classes = len(approxes)
        exponents = np.exp(approxes)
        softmax = exponents / np.sum(exponents)
        targets = np.array([1 if x == targets else 0 for x in range(n_classes)])
        
        der1 = targets - softmax
        der2 = np.zeros((n_classes, n_classes))
        for x in range(n_classes):
            for y in range(n_classes):
                if x == y:
                    der2[x, y] = -softmax[x] * (1 - softmax[y])
                else:
                    der2[x, y] = softmax[x] * softmax[y]
        
        if weights is not None:
            der1 *= weights
            der2 *= weights
        
        return der1, der2
        # return der1, []

train_data = [[1, 4, 5, 6],
              [4, 5, 6, 7],
              [11, 20, 30, 30],
              [30, 40, 50, 60],
              [100, 300, 200, 400]]

train_labels = [0, 0, 1, 1, 2]

train_data = Pool(data=train_data , label=train_labels )

# Initialize CatBoostClassifier with custom `loss_function`
model = CatBoostClassifier(loss_function=MultiClassObjective(),
                           eval_metric="MultiClass",
                           iterations=100,
                           random_seed=0,
                           verbose=True
                          )
# Fit model
model.fit(train_data)
```

Thanks!

**Comment by dimitar-petrov** [comment]
I am also interested in this. @bzografov: Did you managed to fix it.

**Comment by ek-ak** [comment]
Hi! Sorry for very long response. I couldn't reproduce bug in your example. if you can reproduce it, please, post the minimal reproducing example and reopen the issue.
